src/toil/lib/aws/iam.py

```diff
@@ -0,0 +1,140 @@
+
+import logging
+import boto3
+from toil.lib.aws import zone_to_region
+from toil.provisioners.aws import get_best_aws_zone
+from functools import lru_cache
+from typing import Any, List, Dict, Set, cast
+
+from mypy_boto3_iam import IAMClient
+from mypy_boto3_iam.type_defs import GetRolePolicyResponseTypeDef
+from toil.lib.aws.session import AWSConnectionManager
+
+
+logger = logging.getLogger(__name__)
+
+_CLUSTER_LAUNCHING_PERMISSIONS = {"iam:CreateRole",
+                                  "iam:CreateInstanceProfile",
+                                  "iam:TagInstanceProfile",
+                                  "iam:DeleteRole",
+                                  "iam:DeleteRoleProfile",
+                                  "iam:ListAttatchedRolePolicies",
+                                  "iam:ListPolicies",
+                                  "iam:ListRoleTags",
+                                  "iam:PutRolePolicy",
+                                  "iam:RemoveRoleFromInstanceProfile",
+                                  "iam:TagRole"
+                                  }
+
+
+def check_policy_warnings(allowed_actions: Dict[str, List[str]] = {'*': []}, launching_perms: Set[str] = _CLUSTER_LAUNCHING_PERMISSIONS) -> None:
+    """
+    Check whether necessary permissions are permitted for AWS
+
+    :param allowed_actions: Dictionary containing actions allowed by resource
+    :param launching_perms: Set of required actions to launch a cluster on AWS
+    """
+    permissions = [x for x in launching_perms if helper_permission_check(x, allowed_actions["*"])]
+
+    if not launching_perms.issubset(set(permissions)):
+        raise RuntimeError("Missing permissions", permissions)
```

Comment:
This is raising an error and not raising a warning or logging at warning level. So if it appears that we don't have e.g. `iam:DeleteRole` granted on `*`, Toil will bail out and refuse to create the cluster, with no way to override it.

I don't think that's the behavior we want. For one thing, we aren't 100% confident that we're actually testing all the ways a user could have permission to do something, so there will be cases where we do have the permissions we need but we don't know it.

We're also checking for permission grants on `*` (i.e. all resources); there might be ways to grant a permission on less than `*` that will let Toil work, but right now Toil can't understand them and will bail out if it sees them.

We probably want the user to still be able to use Toil in those cases. That could look like some kind of `--force-permissions` flag or something to override the checks, or it could look like just making the checks produce warning-level log messages when they fail, and going on to attempt the things we think we can't do anyway. Or we could pause and make the user hit `y` or something.